src/parser/wakefiles.cpp

```diff
@@ -454,6 +454,64 @@ static std::vector<std::string> filter_wakefiles(std::vector<std::string> &&wake
   return output;
 }
 
+std::vector<std::string> find_external_stdlib_files(bool &ok, bool verbose,
+                                                    const std::string &libdir,
+                                                    FILE *user_warning_dest, const RE2 &exp) {
+  std::vector<std::string> libfiles;
+#ifdef __EMSCRIPTEN__
+  // clang-format off
+    int isNode = EM_ASM_INT({
+      return ENVIRONMENT_IS_NODE;
+    });
+  // clang-format on
+  if (isNode) {
+    if (push_files(libfiles, libdir, exp, 0, user_warning_dest)) ok = false;
+  } else {
+    if (push_packaged_stdlib_files(libfiles, exp, 0)) ok = false;
+  }
+#else
+  if (push_files(libfiles, libdir, exp, 0, user_warning_dest)) ok = false;
+#endif
+  std::sort(libfiles.begin(), libfiles.end());
+  return filter_wakefiles(std::move(libfiles), libdir, verbose);
+}
+
+std::vector<std::string> find_workspace_wakefiles_via_search(bool &ok, bool verbose,
+                                                             const std::string &workdir,
+                                                             FILE *user_warning_dest,
+                                                             const RE2 &exp) {
+  std::vector<std::string> workfiles;
+  if (push_files(workfiles, workdir, exp, 0, user_warning_dest)) ok = false;
+  std::sort(workfiles.begin(), workfiles.end());
+  return filter_wakefiles(std::move(workfiles), workdir, verbose);
+}
+
+std::vector<std::string> find_workspace_wakefiles_via_manifest(const std::string &manifest_path) {
+  std::vector<std::string> workfiles;
+  DiagnosticIgnorer ignorer;
+  ExternalFile file(ignorer, manifest_path.c_str());
+  StringSegment segment = file.segment();
+  std::stringstream in;
+  in.write(reinterpret_cast<const char *>(segment.start), segment.size());
+
+  std::string line;
+  while (std::getline(in, line)) {
+    // strip trailing whitespace (including windows CR)
+    size_t found = line.find_last_not_of(" \t\f\v\n\r");
+    if (found != std::string::npos) {
+      line.erase(found + 1);
```

Comment:
I hate how the string API is just subtly different from all the other container APIs. I had to go look this up to make sure it didn't just remove a single character.